File: 1201-1300/1296-Divide Array in Sets of K Consecutive Numbers/1296-Divide Array in Sets of K Consecutive Numbers.cpp
class Solution {
public:
    bool isPossibleDivide(vector<int>& nums, int k) {
        if (nums.size() % k) {
            return false;
        }
        
        map<int, int> table;
        for (int num : nums) {
            ++table[num];
        }
        
        for (auto p : table) {
            if (p.second > 0) {
                int need = p.second;
                for (int i = p.first; i < p.first + k; ++i) {
                    if (table[i] < need) {
                        return false;
                    }
                    table[i] -= need;
                }
            }
        }
        
        return true;
    }
};

File: 1201-1300/1296-Divide Array in Sets of K Consecutive Numbers/1296-Divide Array in Sets of K Consecutive Numbers.py
import collections
class Solution:
    def isPossibleDivide(self, nums: List[int], k: int) -> bool:
        if len(nums) % k:
            return False

        table = collections.Counter(nums)
        for key in sorted(table):
            if table[key] > 0:
                need = table[key]
                for i in range(key, key + k):                    
                    if table[i] < need:
                        return False
                    table[i] -= need
        return True
